core/src/main/scala/kanaloa/reactive/dispatcher/queue/Worker.scala

```diff
@@ -138,15 +182,21 @@ trait Worker extends Actor with ActorLogging with MessageScheduler {
     }
   }
 
-  private def sendWorkToDelegatee(work: Work, retried: Int, delay: Option[FiniteDuration]): Unit = {
-    val timeoutHandle: Cancellable = delayedMsg(delay.fold(work.settings.timeout)(_ + work.settings.timeout), DelegateeTimeout)
-    maybeDelayedMsg(delay, work.messageToDelegatee, delegatee)
+  private def sendWorkToDelegatee(work: Work, retried: Int): Unit = {
+    val timeoutHandle: Cancellable = delayedMsg(delayBeforeNextWork.fold(work.settings.timeout)(_ + work.settings.timeout), RouteeTimeout)
+    delayBeforeNextWork match {
+      case Some(d) ⇒
+        import context.dispatcher
```

Comment:
It's usually a good idea to make `ExecutionContext` local and explicit. Say if you add another aync operation that demands a dedicated Threadpool, then you might missed the chance if the ExcecutionContext just implicitly read for you. This force you to think in each case which EC to use.